README.md

```diff
@@ -1,2 +1,100 @@
+- GET
+Returns the state of the association resource
+
+- PUT
+Binds the resource pointed to by the given URI(s) to the resource. Return 400 Bad Request if multiple URIs were given for a to-one-association
+
+- POST
+Only supported for collection associations. Adds a new element to the collection.
+
+- DELETE
+Unbinds the association. Return 405 Method Not Allowed if the association is non-optional
+
+### Error codes
+400 Bad Request - if multiple URIs were given for a to-one-association
+
+404 Not found - if the requested entity or collection doesn't exists
+
+405 Method Not Allowed - if the methods is not implemented or a DELETE methods is called on a non-optional association
+
+## HATEOAS & HAL
+The new REST DSpace API supports the HATEOAS paradigm and adopt the HAL format to express links and embedded resources.
+
+## Pagination
+Each endpoints that expose a collection of resources, including sub-paths for embedded or linked collections (aka list of items of a collection, etc.), MUST implement the pagination with the following common behavior
+
+### Request parameter
+- **page**: 0 based integer value that specify the requested page in the result set [default 0]
+- **size**: the dimension of the result set window to show. It must be a positive value. Negative or zero value must be rejected with a 400 Error code. The default value as well as maximum values are configurable by the system administrator. Different Maximum values apply for anonymous users, logged-in users and administrators. Size over the maximum value are automatically reset to the maximum allowed value, no error is thrown.
+- **sort**: the criteria to use. Ordering is specified appending a comma and the keyword asc or desc to the criteria name (i.e. title,asc). Unknown sort criteria and/or ordering keyword produce an error response with Http Code 400
+
+### Response
+The HAL document always includes the following attributes
+
+and, when applicable, the following links
+- **self**: a parameterized link to the requested collection page
+- **next**: the link to the next page of resources in the collection, if any, keeping the same option for size and sorting 
+- **previous**: the link to the previous page of resources in the collection, if any, keeping the same option for size and sorting
+- **first**: the link to the first page of resources in the collection, keeping the same option for size and sorting
+- **last**: the link to the last page of resources in the collection, keeping the same option for size and sorting
+
+### Out of bound pages
+In the case that the request parameters lead to a page outside the result set an empty page should be returned with the links needed to go to the first and last page of the result set if the results set is not empty and the total number of resources in the collection
+
+### Error Code
+400 Bad Request. If an unknown sort criteria is requested or a not valid ordering keyword is specified
+
+## ETags & conditional headers
```

Comment:
What code will be responsible for generating/managing ETags?  Is this provided by the framework, or will DSpace REST need to implement this?  Where will an ETag be provided in a response?  How will one be used in a subsequent request.  An example would be useful here.